File: keycloak/26.7.1/analyzed/core/src/main/java/org/keycloak/representations/info/FeatureRepresentation.java
package org.keycloak.representations.info;

import java.util.Set;

/**
 * Keycloak 服务器功能开关的 REST 表示，用于 {@link ServerInfoRepresentation} 中描述各特性的启用状态与依赖关系。
 */
public class FeatureRepresentation {
    /** 功能内部名称（配置键）。 */
    private String name;
    /** 面向管理员/UI 的显示标签。 */
    private String label;
    /** 功能类型，见 {@link FeatureType}。 */
    private FeatureType type;
    /** 当前是否已启用。 */
    private boolean isEnabled;
    /** 是否已标记为废弃；非 {@code true} 时在 JSON 中省略。 */
    private Boolean deprecated;
    /** 启用本功能前须先启用的其他功能名称集合。 */
    private Set<String> dependencies;

    /** 默认构造函数，供 JSON 反序列化使用。 */
    public FeatureRepresentation() {
    }

    /** @return 功能名称 */
    public String getName() {
        return name;
    }

    /** @param name 功能名称 */
    public void setName(String name) {
        this.name = name;
    }

    /** @return 显示标签 */
    public String getLabel() {
        return label;
    }

    /** @param label 显示标签 */
    public void setLabel(String label) {
        this.label = label;
    }

    /** @return 功能类型 */
    public FeatureType getType() {
        return type;
    }

    /** @param type 功能类型 */
    public void setType(FeatureType type) {
        this.type = type;
    }

    /** @return 是否已废弃 */
    public Boolean isDeprecated() {
        return deprecated;
    }

    /**
     * 设置废弃标记；仅 {@code true} 会保留，否则置为 {@code null} 以省略序列化。
     *
     * @param deprecated 是否废弃
     */
    public void setDeprecated(Boolean deprecated) {
        this.deprecated = Boolean.TRUE.equals(deprecated) ? deprecated : null;
    }

    /** @return 是否已启用 */
    public boolean isEnabled() {
        return isEnabled;
    }

    /** @param enabled 是否启用 */
    public void setEnabled(boolean enabled) {
        isEnabled = enabled;
    }

    /** @return 依赖的功能名称集合 */
    public Set<String> getDependencies() {
        return dependencies;
    }

    /** @param dependencies 依赖的功能名称集合 */
    public void setDependencies(Set<String> dependencies) {
        this.dependencies = dependencies;
    }
}

File: keycloak/26.7.1/analyzed/js/libs/keycloak-admin-client/src/defs/featureRepresentation.ts
/**
 * 服务器特性（Feature）开关的 REST 表示，用于查询或切换预览/实验性功能。
 */
export default interface FeatureRepresentation {
  /** 特性内部名称（如 admin-fine-grained-authz） */
  name: string;
  /** 管理界面或文档中展示的可读标签 */
  label: string;
  /** 特性分类（默认启用、预览、实验性等） */
  type: FeatureType;
  /** 当前是否已启用 */
  enabled: boolean;
  /** 是否已标记为废弃 */
  deprecated?: boolean;
  /** 启用本特性前必须先启用的依赖特性名称列表 */
  dependencies: string[];
}

/** 特性生命周期与默认启用策略的分类枚举 */
export enum FeatureType {
  /** 默认启用且为稳定特性 */
  Default = "DEFAULT",
  /** 默认禁用，需显式开启 */
  DisabledByDefault = "DISABLED_BY_DEFAULT",
  /** 预览特性，默认启用 */
  Preview = "PREVIEW",
  /** 预览特性，默认禁用 */
  PreviewDisabledByDefault = "PREVIEW_DISABLED_BY_DEFAULT",
  /** 实验性特性，可能随时变更或移除 */
  Experimental = "EXPERIMENTAL",
  /** 已废弃特性，建议迁移替代方案 */
  Deprecated = "DEPRECATED",
}
